packages/components/src/components/hds/icon/index.hbs

```diff
@@ -8,21 +8,53 @@
   ...attributes
   aria-hidden="{{if @title 'false' 'true'}}"
   aria-labelledby={{this.ariaLabelledby}}
-  data-test-icon={{@name}}
+  data-test-icon={{this.name}}
   fill="{{this.fillColor}}"
   id={{this._iconId}}
   role={{this.role}}
   width="{{this.svgSize.width}}"
   height="{{this.svgSize.height}}"
-  viewBox="0 0 {{this.size}} {{this.size}}"
+  viewBox={{this.viewBox}}
   xmlns="http://www.w3.org/2000/svg"
+  {{this.loadCarbonIcon this.hdsCarbon.carbonModeEnabled}}
 >
   {{#if @title}}
     <title id={{this._titleId}}>{{@title}}</title>
-    <g role="presentation">
-      <use href="#flight-{{@name}}-{{this.size}}"></use>
-    </g>
+  {{/if}}
+
+  {{#if this.hdsCarbon.carbonModeEnabled}}
+    {{#if (not-eq this.carbonIcon null)}}
+      {{#each this.carbonIcon.content as |node|}}
```

Comment:
[question] is this how it's done in Carbon too? it seems a risk here, because we're not covering all the possible SVG elements (eg. `line`, `ellipse`, `g`, `clipPath`, `mask` etc - there's more)